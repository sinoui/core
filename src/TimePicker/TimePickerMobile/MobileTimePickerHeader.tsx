import React from 'react';
import styled from 'styled-components';
import HourMinuteView from './HourMinuteView';
import AmPmView from './AmPmView';

interface Props {
  /**
   * 12小时制 是否上午
   */
  isAm: boolean;
  /**
   * 是否为小时表盘
   */
  isHourView: boolean;
  /**
   * 小时
   */
  hour: number;
  /**
   * 分钟
   */
  minute: number;
  /**
   * AM 和PM 改变时的回调
   */
  onChangeAm: (isAm: boolean) => void;
  /**
   * 时分视图切换时的回调函数
   */
  onChangeHourOrMinuteView: (isHour: boolean) => void;
}

const MobileTimePickerHeaderWrapper = styled.div`
  display: flex;
  justify-content: center;
  height: 100px;
  width: 100%;
  padding-left: 50px;
  box-sizing: border-box;
  overflow: hidden;
  background-color: ${(props) => props.theme.palette.primary.main};

  & .sinoui-time-picker-mobile-view__header-hour-minute {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    & div:first-child::after {
      content: ':';
    }
  }

  & .sinoui-time-picker-mobile-view__header-am-pm {
    width: 60px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
`;

export default function MobileTimePickerHeader({
  isHourView,
  isAm,
  hour,
  minute,
  onChangeAm,
  onChangeHourOrMinuteView,
}: Props) {
  return (
    <MobileTimePickerHeaderWrapper>
      <div className="sinoui-time-picker-mobile-view__header-hour-minute">
        <HourMinuteView
          value={hour}
          selected={isHourView}
          onChange={() => onChangeHourOrMinuteView(true)}
        />
        <HourMinuteView
          value={minute === 60 ? 0 : minute}
          selected={!isHourView}
          onChange={() => onChangeHourOrMinuteView(false)}
        />
      </div>
      <div className="sinoui-time-picker-mobile-view__header-am-pm">
        <AmPmView selected={isAm} onChange={() => onChangeAm(true)}>
          上午
        </AmPmView>
        <AmPmView selected={!isAm} onChange={() => onChangeAm(false)}>
          下午
        </AmPmView>
      </div>
    </MobileTimePickerHeaderWrapper>
  );
}
