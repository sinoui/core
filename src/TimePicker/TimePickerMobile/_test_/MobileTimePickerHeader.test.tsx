import React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import MobileTimePickerHeader from '../MobileTimePickerHeader';

describe('快照测试', () => {
  it('Clock', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <MobileTimePickerHeader isHourView isAm hour={3} minute={20} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
