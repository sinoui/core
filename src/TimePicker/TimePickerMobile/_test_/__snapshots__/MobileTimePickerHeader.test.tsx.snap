// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 Clock 1`] = `
<div
  className="sc-AxiKw dkguUZ"
>
  <div
    className="sinoui-time-picker-mobile-view__header-am-pm"
  >
    <div
      className="sc-AxirZ fJTPNu"
      onClick={[Function]}
      selected={true}
    >
      上午
    </div>
    <div
      className="sc-AxirZ hqlfic"
      onClick={[Function]}
      selected={false}
    >
      下午
    </div>
  </div>
  <div
    className="sinoui-time-picker-mobile-view__header-hour-minute"
  >
    <div
      className="sc-AxjAm jFVMWz"
      onClick={[Function]}
      selected={true}
    >
      03
    </div>
    <div
      className="sc-AxjAm eJkthF"
      onClick={[Function]}
      selected={false}
    >
      20
    </div>
  </div>
</div>
`;
